File: SWGradientNodeObjC/SWGradientNode_OSX/OSXViewController.m
//
//  ViewController.m
//  SWGradientNode
//
//  Created by NSSimpleApps on 12.06.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "OSXViewController.h"
@import SpriteKit;
#import "StepGradientScene.h"
#import "SWGradientNode.h"

@interface OSXViewController ()

@property (weak) IBOutlet SKView *skView;

@end

@implementation OSXViewController

- (void)viewDidAppear {
    
    [super viewDidAppear];
    
    dispatch_async(dispatch_get_main_queue(), ^{
        
        if (!self.skView.scene) {
            
            StepGradientScene *stepGradientScene = [StepGradientScene sceneWithSize:self.skView.bounds.size];
            stepGradientScene.scaleMode = SKSceneScaleModeAspectFill;//SKSceneScaleModeResizeFill;
            
            [self.skView presentScene:stepGradientScene];
        }
    });
}

- (IBAction)handlePanGesture:(NSPanGestureRecognizer *)sender {
    
    SKView *skView = (SKView *)sender.view;
    
    CGPoint point = NSPointToCGPoint([sender locationInView:skView]);
    
    SKNode *skNode = [skView.scene nodeAtPoint:point];
    
    if ([skNode isKindOfClass:[SWGradientNode class]]) {
        
        SWGradientNode *node = (SWGradientNode *)skNode;
        node.center = NSMakePoint(point.x/skView.frame.size.width, point.y/skView.frame.size.height);
    }
}

- (IBAction)innerRadiusChanged:(NSSlider *)sender {
    
    StepGradientScene *scene = (StepGradientScene *)self.skView.scene;
    
    CGPoint center = CGPointMake(0.5, 0.5);
        
    SWGradientNode *node = (SWGradientNode *)[scene nodeAtPoint:center];
    node.innerRadius = sender.floatValue;
}

@end

File: SWGradientNodeSwift/SWGradientNode_OSX/OSXViewController.swift
//
//  ViewController.swift
//  SWGradientNode
//
//  Created by NSSimpleApps on 05.06.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import Cocoa
import SpriteKit

class OSXViewController: NSViewController {
    
    @IBOutlet weak var skView: SKView!
    
    override func viewDidAppear() {
        
        super.viewDidAppear()
        
        DispatchQueue.main.async { 
            
            let gradientScene = StepGradientScene()
            gradientScene.size = self.skView.bounds.size
            gradientScene.scaleMode = .aspectFill
            
            self.skView.presentScene(gradientScene)
        }
    }
    
    @IBAction func handlePanGesture(_ sender: NSPanGestureRecognizer) {
        
        if let skView = sender.view as? SKView, let scene = skView.scene as? StepGradientScene {
            
            let location = sender.location(in: skView)
            
            let center = NSPoint(x: location.x/skView.bounds.width, y: location.y/skView.bounds.height)
            
            let node = scene.atPoint(location) as! SWGradientNode
            node.center = center
        }
    }
    
    @IBAction func innerRadiusChanged(_ sender: NSSlider) {
        
        if let scene = self.skView.scene as? StepGradientScene {
            
            let center = CGPoint(x: 0.5, y: 0.5)
            
            let node = scene.atPoint(center) as! SWGradientNode
            node.innerRadius = sender.floatValue
        }
    }
}
